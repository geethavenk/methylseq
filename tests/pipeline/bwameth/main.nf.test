nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "bwameth"
    tag "pipeline"

    test("Bwameth Single-End") {
        when {
            params {
                aligner        = "bwameth"
                save_reference = true
                outdir         = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("single_end_software_versions") },
                { assert snapshot(path("$outputDir/bwameth/alignments/samtools_stats/Ecoli.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/Ecoli.stats"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub1.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub1.stats"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub2.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub2.stats"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub3.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub3.stats"),
                            path("$outputDir/bwameth/deduplicated/Ecoli.markdup.sorted.bam"),
                            path("$outputDir/bwameth/deduplicated/Ecoli.markdup.sorted.bam.bai"),
                            path("$outputDir/bwameth/deduplicated/SRR389222_sub1.markdup.sorted.bam"),
                            path("$outputDir/bwameth/deduplicated/SRR389222_sub1.markdup.sorted.bam.bai"),
                            path("$outputDir/bwameth/deduplicated/SRR389222_sub2.markdup.sorted.bam"),
                            path("$outputDir/bwameth/deduplicated/SRR389222_sub2.markdup.sorted.bam.bai"),
                            path("$outputDir/bwameth/deduplicated/SRR389222_sub3.markdup.sorted.bam"),
                            path("$outputDir/bwameth/deduplicated/SRR389222_sub3.markdup.sorted.bam.bai"),
                            path("$outputDir/bwameth/reference_genome/").list(),
                            path("$outputDir/methyldackel/Ecoli.markdup.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/SRR389222_sub1.markdup.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/SRR389222_sub2.markdup.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/SRR389222_sub3.markdup.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/mbias/Ecoli.mbias.txt"),
                            path("$outputDir/methyldackel/mbias/SRR389222_sub1.mbias.txt"),
                            path("$outputDir/methyldackel/mbias/SRR389222_sub2.mbias.txt"),
                            path("$outputDir/methyldackel/mbias/SRR389222_sub3.mbias.txt"),
                            path("$outputDir/qualimap/Ecoli/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub1/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub2/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub3/genome_results.txt")).match("single_end") },
                { assert new File("$outputDir/fastqc/Ecoli_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/Ecoli_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/SRR389222_sub1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/SRR389222_sub2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/SRR389222_sub3_fastqc.html").exists() },
                { assert new File("$outputDir/multiqc/bwameth/multiqc_report.html").exists() },
                { assert new File("$outputDir/trimgalore/logs/Ecoli_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/logs/Ecoli_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/logs/SRR389222_sub1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/logs/SRR389222_sub2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/logs/SRR389222_sub3.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/bwameth/deduplicated/picard_metrics/Ecoli.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/bwameth/deduplicated/picard_metrics/SRR389222_sub1.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/bwameth/deduplicated/picard_metrics/SRR389222_sub2.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/bwameth/deduplicated/picard_metrics/SRR389222_sub3.markdup.sorted.MarkDuplicates.metrics.txt").exists() }
            )
        }
    }

    test("Bwameth Single-End With Index") {
        when {
            params {
                aligner        = "bwameth"
                bwa_meth_index = "https://github.com/nf-core/test-datasets/raw/methylseq/reference/Bwameth_Index.tar.gz"
                methyl_kit     = true
                outdir         = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
            )
        }
    }

    test("Bwameth Single-End RRBS") {
        when {
            params {
                aligner        = "bwameth"
                save_reference = true
                skip_trimming  = true
                rrbs           = true
                save_reference = true
                outdir         = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("single_end_rrbs_software_versions") },
                { assert snapshot(path("$outputDir/bwameth/alignments/samtools_stats/Ecoli.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/Ecoli.stats"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub1.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub1.stats"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub2.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub2.stats"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub3.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222_sub3.stats"),
                            path("$outputDir/bwameth/reference_genome/").list(),
                            path("$outputDir/methyldackel/Ecoli.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/SRR389222_sub1.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/SRR389222_sub2.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/SRR389222_sub3.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/mbias/Ecoli.mbias.txt"),
                            path("$outputDir/methyldackel/mbias/SRR389222_sub1.mbias.txt"),
                            path("$outputDir/methyldackel/mbias/SRR389222_sub2.mbias.txt"),
                            path("$outputDir/methyldackel/mbias/SRR389222_sub3.mbias.txt"),
                            path("$outputDir/qualimap/Ecoli/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub1/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub2/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222_sub3/genome_results.txt")).match("single_end_rrbs") },
                { assert new File("$outputDir/fastqc/Ecoli_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/Ecoli_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/SRR389222_sub1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/SRR389222_sub2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/SRR389222_sub3_fastqc.html").exists() },
                { assert new File("$outputDir/multiqc/bwameth/multiqc_report.html").exists() }
            )
        }
    }

    test("Bwameth Single End RRBS With Index") {
        when {
            params {
                aligner = "bwameth"
                skip_trimming = true
                rrbs = true
                bwa_meth_index = "https://github.com/nf-core/test-datasets/raw/methylseq/reference/Bwameth_Index.tar.gz"
                methyl_kit = true
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
            )
        }
    }

}
