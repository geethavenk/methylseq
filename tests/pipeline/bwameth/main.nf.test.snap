{
    "single_end_software_versions": {
        "content": [
            "{BWAMETH_ALIGN={bwameth=0.2.2}, BWAMETH_INDEX={bwameth=0.2.2}, FASTQC={fastqc=0.12.1}, QUALIMAP_BAMQC={qualimap=2.2.2-dev}, TRIMGALORE={trimgalore=0.6.7, cutadapt=3.4}, Workflow={nf-core/methylseq=v2.7.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-26T22:13:53.261605036"
    },
    "single_end_rrbs_software_versions": {
        "content": [
            "{BWAMETH_ALIGN={bwameth=0.2.2}, BWAMETH_INDEX={bwameth=0.2.2}, FASTQC={fastqc=0.12.1}, QUALIMAP_BAMQC={qualimap=2.2.2-dev}, Workflow={nf-core/methylseq=v2.7.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-26T22:14:55.167421064"
    },
    "single_end": {
        "content": [
            "Ecoli.flagstat:md5,f520e87cc4050799a0dbed72909572ec",
            "Ecoli.stats:md5,c09155913e0164da9010a0840618cf1f",
            "SRR389222_sub1.flagstat:md5,ad8159a7683b9c3057bf54ccb77a2ddc",
            "SRR389222_sub1.stats:md5,047e5fc07dd5975b6fcba41f7e506ef1",
            "SRR389222_sub2.flagstat:md5,e768b4f54014d62b096152c44f920bc1",
            "SRR389222_sub2.stats:md5,145fa2837bbaa85e6daf29e85f57ee79",
            "SRR389222_sub3.flagstat:md5,39516f7d2c77e6fef9543edf8cddbcd6",
            "SRR389222_sub3.stats:md5,1ddf0b5edccc308e1bc90b9a1afa6ca6",
            "Ecoli.markdup.sorted.bam:md5,0d1c814b476521286a829dc64fc3490a",
            "Ecoli.markdup.sorted.bam.bai:md5,befef6ce0a34e2ce7f66c073d9bf4abd",
            "SRR389222_sub1.markdup.sorted.bam:md5,cf2fa2a8b4fed92e3a03afd8999926b4",
            "SRR389222_sub1.markdup.sorted.bam.bai:md5,c95cb45c3f0a9a82bf78a7914b38ebaf",
            "SRR389222_sub2.markdup.sorted.bam:md5,a482e5162b902af2e6f692cdd829686e",
            "SRR389222_sub2.markdup.sorted.bam.bai:md5,fbe5ef44841a8b6028bdb7c14ae67e59",
            "SRR389222_sub3.markdup.sorted.bam:md5,314bb45a56082caced5025f9316a6393",
            "SRR389222_sub3.markdup.sorted.bam.bai:md5,d7d62a85aac95328011ba48c37f3cf1c",
            [
                [
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de",
                    "genome.fa.bwameth.c2t:md5,e51d48ed28fa0c26e2f9c9f13d09403b",
                    "genome.fa.bwameth.c2t.amb:md5,010a242c6764efb30141868a45d698b3",
                    "genome.fa.bwameth.c2t.ann:md5,09b4db3d87a2d4dac9e10e807f377110",
                    "genome.fa.bwameth.c2t.bwt:md5,4646f8ae6bd523b7f4106bbd32eff95e",
                    "genome.fa.bwameth.c2t.pac:md5,a662d7add09698e25c8152fd2306ec66",
                    "genome.fa.bwameth.c2t.sa:md5,4a1e905b828396a1909669bc9c573d8d"
                ]
            ],
            "Ecoli.markdup.sorted_CpG.bedGraph:md5,00132adede8128b3a111136713a2a4ff",
            "SRR389222_sub1.markdup.sorted_CpG.bedGraph:md5,0944ad0e3b05791bf6dfd06be7ca4af1",
            "SRR389222_sub2.markdup.sorted_CpG.bedGraph:md5,0a3c19af55d1625e91903ae25d1d2098",
            "SRR389222_sub3.markdup.sorted_CpG.bedGraph:md5,a91e6b039e9581a9d2ef6f6057918b39",
            "Ecoli.mbias.txt:md5,3a448a2a6364b2de3129365a271c0534",
            "SRR389222_sub1.mbias.txt:md5,fce04d733e066d0b933cedc602e2af81",
            "SRR389222_sub2.mbias.txt:md5,fce04d733e066d0b933cedc602e2af81",
            "SRR389222_sub3.mbias.txt:md5,fce04d733e066d0b933cedc602e2af81",
            "genome_results.txt:md5,d1a1b49086982318f0f94ba7059f71a3",
            "genome_results.txt:md5,2c755b15ae5ad4a1d29395320a6264ca",
            "genome_results.txt:md5,2294badeccd2e8870fd6c60d115687b0",
            "genome_results.txt:md5,4da1144933d094185d02f7482c120804"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-21T22:52:02.181831818"
    },
    "single_end_rrbs": {
        "content": [
            "Ecoli.flagstat:md5,7c3bae0f9c700dde88785a7e94c6ae2c",
            "Ecoli.stats:md5,46eaf2559610c864bc4d27a08a0645b4",
            "SRR389222_sub1.flagstat:md5,897d500a710a56a7098172167fa71108",
            "SRR389222_sub1.stats:md5,0d43bc949dd52241bf6616672bf9409a",
            "SRR389222_sub2.flagstat:md5,37d33d68b31d4550620dc17582a3a750",
            "SRR389222_sub2.stats:md5,d663612293502df7d0cec6c6d99f79dc",
            "SRR389222_sub3.flagstat:md5,e3e925a7b7356e93d14f5e9e102c55b7",
            "SRR389222_sub3.stats:md5,22ece6d2ba9c24b0b7f5d8a49b6e4113",
            [
                [
                    "genome.fa:md5,923a0a268ad29fee3c3437d00f9970de",
                    "genome.fa.bwameth.c2t:md5,e51d48ed28fa0c26e2f9c9f13d09403b",
                    "genome.fa.bwameth.c2t.amb:md5,010a242c6764efb30141868a45d698b3",
                    "genome.fa.bwameth.c2t.ann:md5,09b4db3d87a2d4dac9e10e807f377110",
                    "genome.fa.bwameth.c2t.bwt:md5,4646f8ae6bd523b7f4106bbd32eff95e",
                    "genome.fa.bwameth.c2t.pac:md5,a662d7add09698e25c8152fd2306ec66",
                    "genome.fa.bwameth.c2t.sa:md5,4a1e905b828396a1909669bc9c573d8d"
                ]
            ],
            "Ecoli.sorted_CpG.bedGraph:md5,41f4218573d5e4ff7016ffb42f13a551",
            "SRR389222_sub1.sorted_CpG.bedGraph:md5,4475a9462ce86a019015b69b82a9d5a2",
            "SRR389222_sub2.sorted_CpG.bedGraph:md5,18b8845d9d985d3c203ca6c9c40aab1b",
            "SRR389222_sub3.sorted_CpG.bedGraph:md5,70dcc3836dc9721c2de367d1cf05e815",
            "Ecoli.mbias.txt:md5,ec13da6deba1ce147624bce065ec5df6",
            "SRR389222_sub1.mbias.txt:md5,fce04d733e066d0b933cedc602e2af81",
            "SRR389222_sub2.mbias.txt:md5,fce04d733e066d0b933cedc602e2af81",
            "SRR389222_sub3.mbias.txt:md5,fce04d733e066d0b933cedc602e2af81",
            "genome_results.txt:md5,cd545aa1ee209a1683374bfb5731ab0c",
            "genome_results.txt:md5,f01c68dd46ca9ec0af3d6ec926763b75",
            "genome_results.txt:md5,ef86f3d113782dae1c9bc1660f9f0ae0",
            "genome_results.txt:md5,d7700df4fc6f5b66503ea74d02f1b01b"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-21T22:59:17.759034571"
    }
}